src/services/user/nurseUser.js

```diff
@@ -104,8 +104,34 @@ const getAllNurseUser = async (data, log) => {
   return nurses;
 };
 
+/**
+ * @description The service function that updates a nurse user data
+ * @param {object} data The editable user data and nurseId
+ * @param {function} log Logger utility for logging messages
+ * @returns {object} The updated user data
+ * @throws {Error} Any error that prevents the service from executing successfully
+ */
+const updateNurseUser = async (data, log) => {
+  log.debug('Executing updateNurseUser service');
+  const { name, phoneNo, nurseId } = data;
+  const { NURSE_USER } = db.users.userTypes;
+  const user = await db.users.getUser({ _id: nurseId }, NURSE_USER);
+  if (!user) {
+    log.debug('The nurse user does not exist');
+    const error = new Error('User does not exist');
+    error.httpStatusCode = 404;
+    throw error;
+  }
```

Comment:
We don't need this database trip just to verify that the nurse user exists or not. You can simply attempt updating the nurse resource, if the nurse resource exists, a valid updated document would be returned as response otherwise null would be returned..

Reply:
fixed